drv/vsc7448/src/dev.rs

```diff
@@ -0,0 +1,206 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+use crate::{
+    port::{port10g_flush, port1g_flush},
+    spi::Vsc7448Spi,
+    VscError,
+};
+use vsc7448_pac::Vsc7448;
+
+/// DEV1G and DEV2G5 share the same register layout, so we can write functions
+/// that use either one.
+#[derive(Copy, Clone)]
+pub enum DevGeneric {
+    Dev1g(u32),
+    Dev2g5(u32),
+}
+
+impl DevGeneric {
+    pub fn new_1g(d: u32) -> Self {
+        assert!(d < 24);
```

Comment:
I'd be more comfortable with these asserts if the functions were documented, I'm not sure how to tell if my `d` is valid or why.

Reply:
Added a docstring and switched to returning an error on invalid `d`.